tests/unit/s2n_handshake_misc_test.c

```diff
@@ -48,5 +63,14 @@ int main(int argc, char **argv)
         EXPECT_ERROR_WITH_ERRNO(s2n_handshake_set_finished_len(conn, max_len - 1), S2N_ERR_SAFETY);
     };
 
+    /* Test: s2n_handshake_finish_header */
+    {
+        /* When handshake meesage size is larger than uint16_max */
+        EXPECT_OK(s2n_test_handshake_finish_header(UINT16_MAX + 1));
+
+        /* When handshake message size is larger than uint24_max */
+        EXPECT_ERROR_WITH_ERRNO(s2n_test_handshake_finish_header(UINT24_MAX + 1), S2N_ERR_INTEGER_OVERFLOW);
```

Comment:
Nit: it seems a _little_ silly to define uint24_max using "-1", and then use it with "+1" :)

Rather than defining it and trying to name it, it might be clear enough if you just use (1<<24) directly here. It is a pretty self-documenting constant.